app/code/Magento/ConfigurableProduct/Block/Cart/Item/Renderer/Configurable.php

```diff
@@ -66,7 +66,7 @@ public function getProductForThumbnail()
         if ($this->_scopeConfig->getValue(
             self::CONFIG_THUMBNAIL_SOURCE,
             \Magento\Store\Model\ScopeInterface::SCOPE_STORE
-        ) == ThumbnailSource::OPTION_USE_PARENT_IMAGE ||
+        ) == ThumbnailSource::OPTION_USE_PARENT_IMAGE || (empty($this->getChildProduct()) ||
             !($this->getChildProduct()->getThumbnail() && $this->getChildProduct()->getThumbnail() != 'no_selection')
```

Comment:
I suggest rewriting it in more readable form
```
!($this->getChildProduct() && $this->getChildProduct()->getThumbnail() && $this->getChildProduct()->getThumbnail() != 'no_selection')
```

When `$this->getChildProduct()` can be null by the way?